### DISCUSSION TITLE
Autolink links

### DISCUSSION THREAD

**Discussion by furilo** [outline/outline#1884] [Ideas]
When writing a URL it would be superuseful to have it autolinked so you don't have to do that work manually. So writing this:

<img width="385" alt="imagen" src="https://user-images.githubusercontent.com/24740/107934657-9f0f2900-6f80-11eb-9535-f7a0fa934a36.png">

would give you: 

<img width="265" alt="imagen" src="https://user-images.githubusercontent.com/24740/107934852-dc73b680-6f80-11eb-9c99-62270ad3486c.png">

**Comment by matbrgz** [comment]
Link internal documents will be awesome too. Like Wikipedia when I write a document name it automatic link to existing document.

**Reply by tommoor** [reply]
Don't you have to surround the name with syntax to have it link on Wikipedia? You can link to any internal document with CMD+K, or highlight text and search for the document to link to 😄

**Reply by wvds** [reply]
@tommoor Not sure if this is the correct place to ask, or that I should create another issue/discussion, if so, let me know.

But wouldn't it be nice to have something to autolink specific texts?
As an example that you could configure that ABC-1234 will automatically create a link to https://example.com/ticket/ABC-1234.

I think it can be really powerful, and my company would definitely use it to autolink ticket numbers to their specific Jira url for example.

Mattermost has also a plugin available for this, which could be an inspiration for the implementation: https://github.com/mattermost/mattermost-plugin-autolink

I've already made a local POC within the code of Outline, making use of the Prosemirror plugin system.

What do you think about the feature and do you think it would be useful and usable?

**Reply by tommoor** [reply]
Hi Wesley, it would definitely be cool to have this functionality – I think my main pushback is that I'm struggling to think of a usecase for auto-linking outside of issue numbers (it's also the example usage on the mattermost plugin). If that's the case then we're probably better served in the longrun by prioritizing a real integration for Jira/GitHub/Linear. 

I suppose the argument is that you can't cover all the bases and autolinking patterns might be a good fallback. cc @thenanyu for thoughts

**Reply by wvds** [reply]
@tommoor I agree that an integration would be nicer. I've checked the Outline source and architecture document, but I think it currently only has support for embeds? Since both Jira and Bitbuckets don't provide these I thought this was the next best thing, which is generic but can be quite complex for the users (with regexes and such) as well.

If there's a way to make a different kind of integration, for example something that could use the Jira API, I would maybe be able to contribute/help as well.

**Reply by xargi** [reply]
It would already be a major improvement if creating a link on a selected text searched for a page with the same title, instead of only offering to create a new document

Currently when clicking the button link button:
![image](https://user-images.githubusercontent.com/9206039/121577281-0a01d580-ca32-11eb-87d4-46594e3c09af.png)

The result is this:
![image](https://user-images.githubusercontent.com/9206039/121577408-2867d100-ca32-11eb-8e94-64b12dfbe335.png)

When it could be this:
![image](https://user-images.githubusercontent.com/9206039/121577357-1be37880-ca32-11eb-8548-23dc7eac92c4.png)

**Reply by tommoor** [reply]
Hey @xargi this is a good suggestion, if you want to file an issue we'll get it tackled.

**Reply by tommoor** [reply]
> I've checked the Outline source and architecture document, but I think it currently only has support for embeds?

@wvds that's true, have you seen the "hover card" that appears when you hover the mouse over an internal link? I'm imagining that this canvas can be extensible so that you can for example connect JIRA and then hover over a issue number to see the details of that issue.

**Comment by jaitaiwan** [comment]
One of the uses of autolinking would be for zettlekasten (or second brain) purposes.
